crates/nu-parser/tests/test_lite_parser.rs

```diff
@@ -44,39 +49,63 @@ fn comment_beside() -> Result<(), ParseError> {
 
 #[test]
 fn comments_stack() -> Result<(), ParseError> {
+    // Code:
+    // # this is a comment
+    // # another comment
+    // # def foo bar
```

Comment:
I wonder if for cases like this we should treat the comments as a simple block of comments

Reply:
that is a good point. Should those comments be part of the next command or should they be a separate statement?

Reply:
We might have to play with it a bit to know if that's what we want, but I was thinking they should be part of the next command.

Reply:
Sounds like a good plan. So, we could start saying that all comments above a command should be attached to it. 
What about this case?
```
# comment 1
# comment 2
let a = ( ls |
where name =~ some |   # another comment
each { |file| rm file.name } # final comment
```
or this
```
# comment 1
# comment 2
let a = ( ls 
| where name =~ some    # another comment
| each { |file| rm file.name } # final comment
```

Reply:
@elferherrera I think for the second case they'd attach to the things that come before. For the first case, maybe both attach to the second one?

In practice, you'll only really see the ones that are doc comments I think, and they'll be on the definition parts rather than the expressions.